src/type.ts

```diff
@@ -454,7 +454,7 @@ export class Type {
 					let property = this.getProperty(name);
 
 					// Add Property
-					if (!property) {
+					if (!property || (member.type && property.propertyType !== member.type)) {
```

Comment:
Isn't `member.type` technically a `string`, so that comparison would always come back as different?